iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -370,29 +383,48 @@ cxx::expected<int32_t, IpcChannelError> UnixDomainSocket::createSocket(const Ipc
         }
         else
         {
+            auto closeCall = closeFd(sockfd);
+            if (closeCall.has_error())
+            {
+                return cxx::error<IpcChannelError>(closeCall.get_error());
+            }
             return createErrorFromErrnum(bindCall.getErrNum());
         }
     }
     else
     {
-        // we use a connected socket, this leads to a behavior closer to the message queue (e.g. error if client is
-        // created and server not present)
+        // we use a connected socket, this leads to a behavior closer to the message queue (e.g. error if client
+        // is created and server not present)
         auto connectCall = cxx::makeSmartC(connect,
                                            cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
                                            {ERROR_CODE},
-                                           {},
+                                           {ENOENT},
                                            sockfd,
                                            (struct sockaddr*)&m_sockAddr,
                                            static_cast<socklen_t>(sizeof(m_sockAddr)));
 
-        if (!connectCall.hasErrors())
+        if (connectCall.hasErrors())
         {
-            return cxx::success<int32_t>(sockfd);
+            auto closeCall = closeFd(sockfd);
+            if (closeCall.has_error())
+            {
+                return cxx::error<IpcChannelError>(closeCall.get_error());
+            }
+            return createErrorFromErrnum(connectCall.getErrNum());
```

Comment:
This is weird. Let's say you have an error in your `connectCall` and caused by this you experience also an error maybe in the `close` call. Then we return only the closeCall error and not the connectCall error which led to this trouble. Here in this particular case I would recommend ignoring the error value of the closeCall since it is only a symptom and not the source and always return the `createErrorFromErrnum(connectCall.getErrNum());`

Reply:
You are right. Adapted.